src/app/dashboard/train/components/NewModel.tsx

```diff
@@ -0,0 +1,314 @@
+"use client";
+import JSZip from "jszip";
+import { saveAs } from 'file-saver';
+import React, { ReactHTMLElement, useState, useRef, use, useEffect } from 'react'
+import { CiTrash } from "react-icons/ci"
+import { FiUploadCloud } from "react-icons/fi"
+import { MdKeyboardArrowLeft } from "react-icons/md"
+
+import lighthouse from "@lighthouse-web3/sdk";
+import { processFiles } from "@utils/upload";
+import { compute } from "@utils/compute";
+import axios from "axios";
+
+interface IProps {
+    setPage: (page: number | null) => void
+    setModal: (modal: number | null) => void
+    setTrain: (train: boolean) => void
+    train: boolean
+}
+
+
+
+const NewModel = ({ setPage, setModal, setTrain, train }: IProps) => {
+    const trainScriptInput = useRef<HTMLInputElement | null>(null)
+    const requirementInput = useRef<HTMLInputElement | null>(null)
+    const dataSetInput = useRef<HTMLInputElement | null>(null)
+
+    const [trainScript, setTrainScript] = useState<File | null>(null)
+    const [requirementsScript, setRequirementsScript] = useState<File | null>(null)
+    const [modelName, setModelName] = useState<string>("")
+    const [dataSet, setDataSet] = useState<File[]>([])
+
+
+
+    //truncate file name
+    const truncateFileName = (
+        fileName: string,
+        size = 10
+    ) => {
+        const split = fileName.split('.');
+        const filename = split[0];
+        const extension = split[1];
+        if (filename.length > size) {
+            let currFileName = filename.substring(0, size);
+            return currFileName + '....' + extension;
+        } else {
+            return fileName
+        }
+
+    };
+
+    // regex for file extension
+    const regex = new RegExp("[^.]+$");
+
+    //format file size
+    const formatBytes = (bytes: number, decimals = 2) => {
+        if (!+bytes) return '0 Bytes'
+
+        const k = 1024
+        const dm = decimals < 0 ? 0 : decimals
+        const sizes = ['Bytes', 'KiB', 'MiB', 'GiB', 'TiB', 'PiB', 'EiB', 'ZiB', 'YiB']
+
+        const i = Math.floor(Math.log(bytes) / Math.log(k))
+
+        return `${parseFloat((bytes / Math.pow(k, i)).toFixed(dm))} ${sizes[i]}`
+    }
+
+    //train script
+    const handleTrainScriptDragOver = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+
+    }
+
+    const handleTrainScriptDrop = (e: React.DragEvent<Blob>) => {
+        e.preventDefault()
+        const files = e.dataTransfer.files
+        if (!files || files.length > 1) {
+            console.log("upload one file")
+            return
+        }
+
+        if (files[0].name.match(regex)?.[0] === "ipynb") {
```

Comment:
can u list the train scripts and let files choose.

Reply:
I worked based on the design given